authn/verifier.go

```diff
@@ -29,7 +29,22 @@ func NewIDTokenVerifier(issuer, audience string, keychain JWKProvider) (*idToken
 	}
 
 	return &idTokenVerifier{
-		audience:  audience,
+		audience:  jwt.Audience{audience},
+		keychain:  keychain,
+		issuerURL: issuerURL,
+	}, nil
+}
+
+// NewIDTokenVerifierWithAudiences creates a new idTokenVerifier object by using keychain as the JWT provider
+// Claims are verified against issuer and the set of audiences
+func NewIDTokenVerifierWithAudiences(issuer string, audiences jwt.Audience, keychain JWKProvider) (*idTokenVerifier, error) {
```

Comment:
i'm not sure if anyone depends on `NewIDTokenVerifier` being a public function, but i'd advocate for making this new variation private or internal until a requirement presents itself.

Reply:
I agree with waiting for a real requirement of that function being exported. Though removing `NewIDTokenVerifier` is dangerous if someone really relies on it. In the last commit I still changed the response type from `*idTokenVerifier` to `JWTClaimsExtractor` as Go discourage returning unexported types from exported functions.